### PR TITLE
feat: add package filtering

### PR DESCRIPTION
**Pull request by zardak**

### CODE REVIEW

**Comment by spalt08** [comment]
image

File: packages/web/src/components/ui/ProblemsList/ProblemsList.tsx
Comment (spalt08) [inline packages/web/src/components/ui/ProblemsList/ProblemsList.tsx:9]: nit: `onChange`

File: packages/web/src/components/ui/SearchResultsSidebar/SearchResultsSidebar.tsx
Comment (spalt08) [inline packages/web/src/components/ui/SearchResultsSidebar/SearchResultsSidebar.tsx:127]: I guess it should be `onReturnClick` and `onResetClick`

File: packages/web/src/components/ui/SearchResultsSidebar/SearchResultsSidebar.tsx
Comment (ctizen) [inline packages/web/src/components/ui/SearchResultsSidebar/SearchResultsSidebar.tsx:141]: Looks like renaming the 'problems' entity to 'traits' was unnecessary. However, my point is - we shouldn't use different terminology for the same entity, so I probably could agree with renaming to 'traits' _everywhere_, but now it's not the case.

File: packages/web/src/store/selectors/websiteResults.ts
Comment (spalt08) [inline packages/web/src/store/selectors/websiteResults.ts:57]: I guess that may conflict with #80

File: packages/web/src/store/selectors/scanDisplayOptions/packageFilters.ts
Comment (ctizen) [inline packages/web/src/store/selectors/scanDisplayOptions/packageFilters.ts:4]: Are there non-optimized filters somewhere?

File: packages/web/src/store/selectors/scanDisplayOptions/scanDisplayOptions.ts
Comment (spalt08) [inline packages/web/src/store/selectors/scanDisplayOptions/scanDisplayOptions.ts:33]: Why do you need to have both optimized and non-optimized scan display options? Can't we use sets everywhere?

File: packages/web/src/components/ui/KeywordsList/KeywordsList.tsx
Comment (zardak) [reply packages/web/src/components/ui/KeywordsList/KeywordsList.tsx:10]: Valid point, but components are in need of much heavier pass of refactoring.

File: packages/web/src/store/selectors/scanDisplayOptions/scanDisplayOptions.ts
Comment (zardak) [reply packages/web/src/store/selectors/scanDisplayOptions/scanDisplayOptions.ts:33]: That's in the plans for refactoring pass.

File: packages/web/src/components/ui/SearchResultsSidebar/SearchResultsSidebar.tsx
Comment (zardak) [reply packages/web/src/components/ui/SearchResultsSidebar/SearchResultsSidebar.tsx:141]: I see where you're coming from, but distinct names in business logic domain and UI presentation layer aren't a problem per se and encountered from time to time. In this very case I'd still resort to current naming having imminent additional features (like `duplicate`, etc.) in mind.

File: packages/web/src/store/selectors/scanDisplayOptions/packageFilters.ts
Comment (zardak) [reply packages/web/src/store/selectors/scanDisplayOptions/packageFilters.ts:4]: In the store itself, in raw serializable form.

File: packages/web/src/store/selectors/websiteResults.ts
Comment (zardak) [reply packages/web/src/store/selectors/websiteResults.ts:57]: Updated to use this field across the component.

File: packages/web/src/components/ui/SearchResultsSidebar/SearchResultsSidebar.tsx
Comment (ctizen) [reply packages/web/src/components/ui/SearchResultsSidebar/SearchResultsSidebar.tsx:141]: Different naming increases cognitive load and amounts of chaos in the codebase. It's not about logic domain or ui, it's about how you or someone else reads the code and understands its meaning and how different parts are connected to each other. I can approve this if you insist, though it is to be refactored in nearest future.

**Review by spalt08 — APPROVED** [review]

**Review by ctizen — APPROVED** [review]

File: packages/web/src/components/ui/SearchResultsSidebar/SearchResultsSidebar.tsx
Comment (zardak) [reply packages/web/src/components/ui/SearchResultsSidebar/SearchResultsSidebar.tsx:141]: The reasoning you gave was exactly the case why i chose to rename it the current way. "Package trait" or "Package feature" when talking about a "vulnerable" / "outdated" / "duplicate" package makes much more sense to me rather than a "Package problem". @spalt08 Your call.

**Pull request merged**
